src/components/KCatalog/KCatalog.vue

```diff
@@ -486,7 +486,9 @@ export default defineComponent({
       hasInitialized.value = true
     }
 
-    const { query, search } = useDebounce('', 350)
+    const query = ref('')
+    const [search] = useDebounce((q: string) => { query.value = q })
```

Comment:
This appears to be a breaking change since the return format is now an array instead of an object?

Reply:
Yes. We only have two components using this composable: KTable and KCatalog. I intended to make `useDebounce` return two functions: one is the denounced function, and the other one is a function that accepts a delay (other than the default one) and returns a new debounced function with that new delay (but which shares the same timeout handle). I tried to implement this composable to make it as general as possible for different circumstances.

This allows us to trigger the underlying function with a different delay. For example, if we want to call the function immediately and clear all pending calls, we can use the second function to generate a debounced function with no delay and trigger it.

Reply:
Is it possible to refactor the composable to return an object instead of an array; one property for each function?

Reply:
I refactored the `useDebounce` composable to make it returns an object instead of an array.